overlord/stateengine.go

```diff
@@ -147,7 +147,7 @@ func (se *StateEngine) Ensure() error {
 	for _, m := range se.managers {
 		err := m.Ensure()
 		if err != nil {
-			logger.Noticef("state ensure error: %v", err)
+			logger.Noticef("state ensure error from %T: %v", m, err)
```

Comment:
devicemgr has it's own ensureError which adds `devicemgr: ..` prefix to the error string. Maybe we could do something like this:

```
diff --git a/overlord/stateengine.go b/overlord/stateengine.go
index 80b2755c4d..f4da008bdc 100644
--- a/overlord/stateengine.go
+++ b/overlord/stateengine.go
@@ -121,6 +121,14 @@ func (se *StateEngine) StartUp() error {
 type ensureError struct {
        errs []error
 }
+type managerEnsureError struct {
+       manager string
+       err     error
+}
+
+func (m *managerEnsureError) Error() string {
+       return fmt.Sprintf("%s: %v", m.manager, m.err)
+}
 
 func (e *ensureError) Error() string {
        return fmt.Sprintf("state ensure errors: %v", e.errs)
@@ -147,8 +155,12 @@ func (se *StateEngine) Ensure() error {
        for _, m := range se.managers {
                err := m.Ensure()
                if err != nil {
-                       logger.Noticef("state ensure error: %v", err)
-                       errs = append(errs, err)
+                       merr := &managerEnsureError{
+                               manager: fmt.Sprintf("%T", m),
+                               err:     err,
+                       }
+                       logger.Noticef("state ensure error: %v", merr)
+                       errs = append(errs, merr)
                }
        }
        if len(errs) != 0 {
```

Note that the error strings are not too nice:
```
state ensure errors: [*overlord_test.fakeManager: boom1 *overlord_test.fakeManager: boom2]
```